src/background/services/wallet/WalletService.ts

```diff
@@ -593,21 +593,39 @@ export class WalletService implements OnUnlock {
       throw new Error('Wallet not found');
     }
 
-    if (isSolanaSigningRequest(tx)) {
+    if (isSolanaRequest(tx)) {
       if (
         !(wallet instanceof SolanaSigner) &&
         !(wallet instanceof SolanaLedgerSigner)
       ) {
         throw new Error('Unable to find a proper signer');
       }
 
-      const signedTx = await wallet.signTx(
-        tx.data,
-        (await getProviderForNetwork(network)) as SolanaProvider,
-      );
+      if (isSolanaMsgRequest(tx)) {
+        if (wallet instanceof SolanaLedgerSigner) {
+          /**
+           * FIXME:
+           * I have a PoC that seems to be working, but obtained signatures are not verified
+           * properly by the dApps. I think it's because the dApps provide a UTF-8 message,
+           * but for it to be accepted by Solana Ledger app, we need to serialize it,
+           * add a message header etc., and I think Ledger then signs the whole thing, which
+           * makes it impossible to verify the signature with the original message.
+           */
```

Comment:
There is also [a PR opened to the Solana App repo 2 days ago](https://github.com/LedgerHQ/app-solana/pull/118) that may fix it (but I'm not sure, there's no description and it has 500+ files changed, so my macbook gives up 😆).